js/service.js

```diff
@@ -121,15 +121,16 @@ function callURL(route) {
 }
 
 /**
- * Make an POST XHR request and return a promise to resolve it.
+ * Make a POST or PUT XHR request and return a promise to resolve it.
  * @param  {Object} params A key-value object to be posted
  * @returns {Promise} A promise to resolve an XHR request
  */
-function postEntry(entryData) {
+function updateEntry(requestType = ``, endpointRoute, entryData) {
```

Comment:
we'll need to update the code in this function so that it considers all HTTP `20x` as "things worked" - with the moderation PUT operation returning a 204 code ("everything okay, there just isn't any data to send back to you") the current code treats that 204 as "not 200" and so it mistakenly identifies the code as an error